test/cg.jl

```diff
@@ -50,6 +50,37 @@ Random.seed!(1234321)
         x0 = cg(A, zeros(T, n))
         @test x0 == zeros(T, n)
     end
+
+    @testset "Matrix{$T}, unconjugated dot product" for T in (Float32, Float64, ComplexF32, ComplexF64)
+        A = rand(T, n, n)
+        A = A + transpose(A) + 15I
+        x = ones(T, n)
+        b = A * x
+
+        reltol = √eps(real(T))
+
+        # Solve without preconditioner
+        x1, his1 = cocg(A, b, reltol = reltol, maxiter = 100, log = true)
+        @test isa(his1, ConvergenceHistory)
+        @test norm(A * x1 - b) / norm(b) ≤ reltol
+
+        # With an initial guess
+        x_guess = rand(T, n)
+        x2, his2 = cocg!(x_guess, A, b, reltol = reltol, maxiter = 100, log = true)
+        @test isa(his2, ConvergenceHistory)
+        @test x2 == x_guess
+        @test norm(A * x2 - b) / norm(b) ≤ reltol
+
+        # The following tests fails CI on Windows and Ubuntu due to a
+        # `SingularException(4)`
+        if T == Float32 && (Sys.iswindows() || Sys.islinux())
+            continue
+        end
+        # Do an exact LU decomp of a nearby matrix
+        F = lu(A + rand(T, n, n))
+        x3, his3 = cocg(A, b, Pl = F, maxiter = 100, reltol = reltol, log = true)
+        @test norm(A * x3 - b) / norm(b) ≤ reltol
```

Comment:
For tests like this you can use `≈`.  It should be equivalent to:
```jl
@test A*x3 ≈ b rtol=reltol
```

Reply:
Hmmm.  I cannot seem to specify the keyword arguments of `isapprox` in its infix form:
```
julia> VERSION
v"1.5.4-pre.0"

julia> 0 ≈ 0
true

julia> 0 ≈ 0, rtol=1e-8
ERROR: syntax: "0" is not a valid function argument name around REPL[23]:1
```
Is this a new capability introduced in version > 1.5.4?

Reply:
Don't use the comma, and have it in a `@test`:
```julia
julia> @test 0 ≈ 0 rtol=1e-8
Test Passed
```